services-gateway-netty/src/main/java/io/scalecube/services/gateway/SessionEventHandler.java

```diff
@@ -16,7 +21,7 @@
    * @param req request message (not null)
    * @return message
    */
-  default GatewayMessage mapMessage(WebsocketSession session, GatewayMessage req) {
+  default M mapMessage(GatewaySession session, M req) {
```

Comment:
`GatewaySession` can also be made as typed parameter, same as with `M`.

Reply:
that was the initial version... 2 generics.

 @dmytro-lazebnyi suggested to go for WebsocketSession interface, and actually I like it also

Reply:
I mean make `GatewaySession` as `S` as type parameter. Same as with `M`. I.e implementation would set its  own concrete `GatewaySession` type and in method param impl. would deal with concrete type.